### PR TITLE
Ability to access parent and child information associated with a ViewHolder

### PR DESCRIPTION
**Pull request by paul-turner**
Had to do some package reshuffling so this will likely mean 3.0.0

Tries to address the needs of #157 and other associated issues.

ParentViewHolder now has:
`getParentListItem()`
`getAdapterParentPosition()`

ChildViewHolder now has:
`getChildListItem()`
`getAdapterParentPosition()`
`getAdapterChildPosition()`

Still TODO:
- [x] javadocs
- [x] version bump
- [x] snapshot info

### CODE REVIEW

**Comment by paul-turner** [comment]
This is good to go for a `SNAPSHOT` release

**Comment by dgreenhalgh** [comment]
`getAdapterParentPosition()` -> `getParentAdapterPosition()`
`getAdapterChildPosition()` -> `getChildAdapterPosition()`

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/ChildViewHolder.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package com.bignerdranch.expandablerecyclerview;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+/**
+ * ViewHolder for a child list
+ * item.
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/ChildViewHolder.java]: :eyes:

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/ExpandableRecyclerAdapter.java [hunk 2]
```diff
@@ -608,32 +611,52 @@ private void collapseParentListItem(ParentWrapper parentWrapper, int parentIndex
...
-                expandedCount++;
+            if (listItem instanceof ParentWrapper) {
+                parentCount++;
+            }
+        }
+        return parentCount;
+    }
+
+    /**
+     * Given the index relative to the entire recycler view for a child item,
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/ExpandableRecyclerAdapter.java]: /s/recycler view/RecyclerView

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/ParentViewHolder.java [hunk 3]
```diff
@@ -50,6 +55,35 @@ public ParentViewHolder(View itemView) {
     }
 
     /**
+     * @return the ParentListItem associated with this view holder
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/ParentViewHolder.java]: /s/view holder/ViewHolder

File: sampleapp/src/main/java/com/ryanbrooks/expandablerecyclerviewsample/linear/horizontal/HorizontalChildViewHolder.java [hunk 4]
```diff
@@ -25,6 +28,12 @@ public HorizontalChildViewHolder(View itemView) {
         super(itemView);
 
         mDataTextView = (TextView) itemView.findViewById(R.id.list_item_horizontal_child_textView);
+        itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.d(TAG, "Child at " + getAdapterChildPosition() + ", with parent at " + getAdapterParentPosition() + " clicked, child item is \"" + getChildListItem() + "\"");
```
Comment (dgreenhalgh) [inline sampleapp/src/main/java/com/ryanbrooks/expandablerecyclerviewsample/linear/horizontal/HorizontalChildViewHolder.java]: long line

**Pull request merged**